File: Platform/mo-1-common/src/lang-java/org/mo/com/system/FListeners.java
package org.mo.com.system;

import org.mo.com.lang.FObjects;

//============================================================
// <T>监听器集合。</T>
//============================================================
public class FListeners
      extends FObjects<IListener>
{
   //============================================================
   // <T>构造监听器集合。</T>
   //============================================================
   public FListeners(){
      super(IListener.class);
   }

   //============================================================
   // <T>执行处理。</T>
   //
   // @param sender 发送者
   // @param command 命令
   // @param params 参数
   // @return 执行结果
   //============================================================
   public boolean process(Object sender,
                          int command,
                          Object params){
      if(_count > 0){
         for(int n = 0; n < _count; n++){
            if(!_items[n].process(sender, command, params)){
               return false;
            }
         }
      }
      return true;
   }
}

File: Source/Script/MsCommon/System/FListeners.cs
using System;
using MS.Common.Lang;

namespace MS.Common.System
{
   //============================================================
   // <T>监听器列表。</T>
   //
   // @history MAOCY 140414
   //============================================================
   public class FListeners : FObjects<IListener>, IDisposable
   {
      // 监听器列表名称
      public const string NAME = "Listeners";

      //============================================================
      // <T>执行处理。</T>
      //
      // @param sender 发出者
      // @param args 参数列表
      //============================================================
      public void Process(object sender, params object[] args) {
         foreach(IListener listener in this) {
            listener.Process(args);
         }
      }

      #region IDisposable implements

      //============================================================
      // <T>释放处理。</T>
      //============================================================
      public void Dispose() {
         foreach(FLoggerListener listener in this) {
            listener.Dispose();
         }
      }

      #endregion
   }
}
